include/deal.II/distributed/cell_data_transfer.h

```diff
@@ -0,0 +1,346 @@
+     * freedom defined on a parallel::distributed::Triangulation.
+     *
+     * This class has been designed to operate on any kind of datatype that is
+     * serializable. A non-distributed Vector container holding the cell by cell
+     * data has to be provided. Its size has to correspond to
+     * Triangulation::n_active_cells().
+     *
+     * <h3>Transferring cell by cell data</h3>
+     *
+     * The following code snippet demonstrates how to transfer cell-related data
+     * across refinement/coarsening of the registered triangulation.
+     *
+     * @code
+     * // prepare the triangulation,
+     * triangulation.prepare_coarsening_and_refinement();
+     *
+     * // prepare the CellDataTransfer object for coarsening and refinement
+     * // and give the cell data vector that we intend to unpack later,
+     * Vector<double> data_to_transfer (triangulation.n_active_cells());
+     * //[fill data_to_transfer with cell by cell values...]
+     *
+     * parallel::distributed::CellDataTransfer<dim, spacedim, Vector<double>>
+     *   cell_data_trans(triangulation);
+     * cell_data_trans.prepare_for_coarsening_and_refinement(data_to_transfer);
+     *
+     * // actually execute the refinement,
+     * triangulation.execute_coarsening_and_refinement();
+     *
+     * // unpack transferred data,
+     * Vector<double> transferred_data (triangulation.n_active_cells());
+     * cell_data_trans.unpack(transferred_data);
+     *
+     * @endcode
+     *
+     *
+     * <h3>Use for serialization</h3>
+     *
+     * This class can be used to serialize and later deserialize a distributed
+     * mesh with attached data to separate files. If you use more than one
+     * parallel::distributed::CellDataTransfer object, they need to be
+     * serialized and deserialized in the same order.
+     *
+     * For serialization, the following code snippet saves not only the
+     * triangulation itself, but also the cell by cell data attached:
+     * @code
+     * Vector<double> data_to_transfer (triangulation.n_active_cells());
+     * //[fill data_to_transfer with cell by cell values...]
+     *
+     * parallel::distributed::CellDataTransfer<dim, spacedim, Vector<double>>
+     *   cell_data_trans(triangulation);
+     * cell_data_trans.prepare_serialization(data_to_transfer);
+     *
+     * triangulation.save(filename);
+     * @endcode
+     *
+     * Later, during deserialization, both triangulation and data can be
+     * restored as follows:
+     * @code
+     * //[create coarse mesh...]
+     * triangulation.load(filename);
+     *
+     * parallel::distributed::CellDataTransfer<dim, spacedim, Vector<double>>
+     *   cell_data_trans(triangulation);
+     * Vector<double> transferred_data (triangulation.n_active_cells());
+     * cell_data_trans.deserialize(transferred_data);
+     * @endcode
+     *
+     * @note If you use more than one object to attach data to a
```

Comment:
say "more than one CellDataTransfer object" to make it clear what kind of object you're talking about

Reply:
or do you want to be more general here?

Reply:
Any object using the `register_data_attach` and `notify_ready_for_unpack` interface. I know that`SolutionTransfer` and `ParticleHandler` are using it. I can be more specific on that.